hg-fast-export.py

```diff
@@ -388,9 +388,10 @@ def export_note(ui,repo,revision,count,authors,encoding,is_first):
   wr()
   return checkpoint(count)
 
-def export_tags(ui,repo,old_marks,mapping_cache,count,authors,tagsmap):
+def export_tags(ui,repo,old_marks,mapping_cache,count,authors,tagsmap,tag_encoding):
   l=repo.tagslist()
   for tag,node in l:
+    tag=tag.decode(tag_encoding).encode('utf8')
```

Comment:
It would be take default global encoding, this is full analog, as fn_encoding.

```python
  tag_encoding=encoding
  if options.tag_encoding!=None:
    tag_encoding=options.tag_encoding
```

Reply:
If the user does not give an `-e <encoding>` flag, the global `encoding` will be `''`, `decode`  does not take kindly to be called with the empty string and will report an error. That is why the other decode-calls are guarded.

Reply:
Thanks, that fixed